graph/nodecolorer.cpp

```diff
@@ -76,8 +76,13 @@ QColor DepthNodeColorer::get(const GraphicsItemNode *node) {
 
     double lowValue = g_settings->lowDepthValue, highValue = g_settings->highDepthValue;
     if (g_settings->autoDepthValue) {
-        lowValue = m_graph->m_firstQuartileDepth;
-        highValue = m_graph->m_thirdQuartileDepth;
+        if (m_graph->currentScope().graphScope() == DEPTH_RANGE) {
```

Comment:
So, instead of fetching the scope from the graph (which is not really something belongs to it), it would be better to route it through `DepthNodeColorer`, or rather calculate low / high value when colorer is created.